### DISCUSSION TITLE
Server-side filtering. How to convert the column filter state into the format that the backend is excepting?

### DISCUSSION THREAD

**Discussion by its-monotype** [KevinVandy/material-react-table#324] [Q&A]
I'm trying to implement server-side column filtering for a table in my React app, but I'm having trouble converting the column filters state into the format that the backend (which uses the "[nestjs-paginate](https://www.npmjs.com/package/nestjs-paginate)" library) is expecting.

One solution could be to use the "filterFn" property for each column and pass in the filter operator that the backend is expecting (e.g. '$eq', '$gt', etc.) along with the value.
Another approach would be to define a separate mapping function that maps the "ColumnFiltersState" object into the format that the backend is expecting, using the appropriate filter operator and value for each column, but then how we would know which operator to use, maybe add custom meta prop to the "coulmnDef".

Can anyone provide some guidance on how to correctly map the column filters for the backend, and which solution would be the best approach? I would really appreciate any feedback or advice on the best approach, and even better if there is an example code to help me understand the solution better.

**Comment by KevinVandy** [comment]
Just do it you whatever function is sending your api calls.

**Reply by its-monotype** [reply]
But from the column filter state, we can get only the column id and the value, how we can determine which filter operator we need to send to the backend?

**Reply by KevinVandy** [reply]
ah yes, there is a separate `columnFilterFns` state that keeps track of which filterFn is being used. You can see all the states available in the [state options docs](https://www.material-react-table.com/docs/api/state-options)

<img width="734" alt="image" src="https://user-images.githubusercontent.com/28243511/215334573-2a2c9f93-ab2c-42a7-83b8-714498cb86a7.png">
